clang-tools-extra/clang-tidy/add_new_check.py

```diff
@@ -53,7 +55,18 @@ def adapt_cmake(module_path, check_name_camel):
 
 
 # Adds a header for the new check.
-def write_header(module_path, module, namespace, check_name, check_name_camel):
+def write_header(
+    module_path, module, namespace, check_name, check_name_camel, lang_restrict
+):
+    if lang_restrict:
+        override_supported = """
+  bool isLanguageVersionSupported(const LangOptions &LangOpts) const override {
```

Comment:
would be nice to add also getCheckTraversalKind with a TK_IgnoreUnlessSpelledInSource used by default